crates/re_data_store/src/test_util.rs

```diff
@@ -61,3 +63,33 @@ pub fn sanity_unwrap(store: &DataStore) {
         err.unwrap();
     }
 }
+
+// We very often re-use RowIds when generating test data.
+pub fn insert_table_with_retries(store: &mut DataStore, table: &DataTable) {
+    for row in table.to_rows() {
+        let mut row = row.unwrap();
+        loop {
+            match store.insert_row(&row) {
+                Ok(_) => break,
+                Err(WriteError::ReusedRowId(_)) => {
+                    row.row_id = row.row_id.next();
+                }
+                err @ Err(_) => err.map(|_| ()).unwrap(),
+            }
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn init_logs() {
+    use std::sync::atomic::{AtomicBool, Ordering};
+
+    static INIT: AtomicBool = AtomicBool::new(false);
+
+    if INIT
+        .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+        .is_ok()
+    {
+        re_log::setup_native_logging();
```

Comment:
The above code would be more idiomatic using `std::sync::Once`.

It could also be moved into `setup_native_logging`

Reply:
Yeah i've told myself I'd clean this a million times... I'll do so in a follow up.

Reply:
- https://github.com/rerun-io/rerun/issues/4893